tests/unit/deposit/test_zenodo_api_quickstart.py

```diff
@@ -0,0 +1,120 @@
+            )
+            assert res.status_code == 401
+
+            # try get deposits as logged-in user
+            res = client.get(
+                url_for('invenio_deposit_rest.dep_list'),
+                headers=oauth2_headers_user_1
+            )
+            assert res.status_code == 200
+            data = json.loads(res.data.decode('utf-8'))
+            assert data == []
+
+            # create a new deposit
+            deposit = {
+            }
+
+            res = client.post(
+                url_for('invenio_deposit_rest.dep_list'),
+                headers=oauth2_headers_user_1,
+                data=json.dumps(deposit)
+            )
+            assert res.status_code == 201
+            data = json.loads(res.data.decode('utf-8'))
+            deposit_id = data['id']
+            # FIXME
+            #  assert data['files'] == []
+            #  assert data['title'] == ''
+            assert 'created' in data
+            assert 'modified' in data
+            assert 'id' in data
+            assert 'metadata' in data
+            # FIXME
+            #  assert data['state'] == 'unsubmitted'
+            assert data['owner'] == write_token['token'].user_id
+
+            sleep(2)
+
+            # upload a file
+            files = {'file': (BytesIO(b'1, 2, 3'), "myfirstfile.csv"),
+                     'name': 'myfirstfile.csv'}
+
+            res = client.post(
+                data['links']['files'],
+                headers=oauth2_headers_user_1,
+                data=files,
+                content_type='multipart/form-data',
+            )
+            assert res.status_code == 201
+            data = json.loads(res.data.decode('utf-8'))
+            assert data['checksum'] == 'md5:66ce05ea43c73b8e33c74c12d0371bc9'
+            assert data['filename'] == 'myfirstfile.csv'
+            assert data['filesize'] == 7
+            assert data['id']
+
+            # modify deposit
+            deposit = {
+                "metadata": {
+                    "title": "My first upload",
+                    "upload_type": "poster",
+                    "description": "This is my first upload",
+                    "creators": [
+                        {
+                            "name": "Doe, John",
+                            "affiliation": "Zenodo"
+                        }
+                    ]
+                }
+            }
+            res = client.put(
+                url_for('invenio_deposit_rest.dep_item',
+                        pid_value=deposit_id),
+                headers=oauth2_headers_user_1,
+                data=json.dumps(deposit)
+            )
+            assert res.status_code == 200
+
+            # publish deposit
+            res = client.post(
+                url_for('invenio_deposit_rest.dep_actions',
+                        pid_value=deposit_id, action='publish'),
+                headers=oauth2_headers_user_1,
+            )
+            assert res.status_code == 202
```

Comment:
Here it would be good with a last check that the record was created - i.e. get the ``record_id`` from the response data and make a query.